frontend/class-frontend.php

```diff
@@ -915,7 +915,7 @@ private function generate_canonical() {
 				$canonical = get_search_link();
 			}
 			elseif ( is_front_page() ) {
-				$canonical = home_url();
+				$canonical = home_url( user_trailingslashit( '' ) );
```

Comment:
Let's turn this around and have `user_trailingslashit` use `home_url` as it's argument.
`home_url` uses `get_home_url` which does an `ltrim` on the path, which in this case would be `/` resulting in no trailing slash on the url.